src/tests/shamsolvergraph/LifetimeTracker_tests.cpp

```
@@ -0,0 +1,296 @@
+
+            shamrock::solvergraph::LifetimeTracker<INode>::on_create       = &on_create_node;
+            shamrock::solvergraph::LifetimeTracker<INode>::on_destroy      = &on_destroy_node;
+            shamrock::solvergraph::LifetimeTracker<INode>::on_state_update = &on_state_update_node;
+            shamrock::solvergraph::LifetimeTracker<INode>::on_event        = &on_event_node;
+
+            shamrock::solvergraph::LifetimeTracker<IEdge>::on_create       = &on_create_edge;
+            shamrock::solvergraph::LifetimeTracker<IEdge>::on_destroy      = &on_destroy_edge;
+            shamrock::solvergraph::LifetimeTracker<IEdge>::on_state_update = &on_state_update_edge;
+            // LifetimeTracker<IEdge>::on_event is intentionally left null: edges have no
+            // "operation" concept, so it must never be invoked. If it ever is, this test
+            // crashes on a null function-pointer call instead of silently passing.
```

Cppcheck (2.21.0)

[style] The function 'assert_equal_array' is never used.

(unusedFunction)